packages/atomic/src/components/atomic-folded-result-list/atomic-folded-result-list.tsx

```diff
@@ -124,197 +112,66 @@ export class AtomicFoldedResultList implements InitializableComponent {
   @Method() public async setRenderFunction(
     render: (result: FoldedResult) => HTMLElement
   ) {
-    this.renderingFunction = render;
-  }
-
-  private listWrapperRef?: HTMLDivElement;
-
-  private get fields() {
-    if (this.fieldsToInclude.trim() === '') return [];
-    return this.fieldsToInclude.split(',').map((field) => field.trim());
-  }
-
-  private get defaultFieldsToInclude() {
-    return [
-      'date',
-      'author',
-      'source',
-      'language',
-      'filetype',
-      'parents',
-      'ec_price',
-      'ec_name',
-      'ec_description',
-      'ec_brand',
-      'ec_category',
-      'ec_item_group_id',
-      'ec_shortdesc',
-      'ec_thumbnails',
-      'ec_images',
-      'ec_promo_price',
-      'ec_in_stock',
-      'ec_cogs',
-      'ec_rating',
-    ];
+    this.resultListCommon.renderingFunction = render as ResultRenderingFunction;
   }
 
-  public initialize() {
-    this.resultTemplatesManager = buildResultTemplatesManager(
-      this.bindings.engine
+  @Listen('scroll', {target: 'window'})
+  handleInfiniteScroll() {
+    this.resultListCommon.handleInfiniteScroll(
+      this.enableInfiniteScroll,
+      this.host,
+      this.resultList
     );
-    this.resultList = this.initFolding({
-      fieldsToInclude: [...this.defaultFieldsToInclude, ...this.fields],
-    });
-    this.resultsPerPage = buildResultsPerPage(this.bindings.engine);
-    this.registerDefaultResultTemplates();
-    this.registerChildrenResultTemplates();
   }
 
-  private initFolding(options = {}) {
-    const opts: FoldedResultListProps = {};
-    opts.options = {...options};
-    opts.options.folding = {};
-    if (this.collectionField) {
-      opts.options.folding.collectionField = this.collectionField;
-    }
-    if (this.parentField) {
-      opts.options.folding.parentField = this.parentField;
-    }
-    if (this.childField) {
-      opts.options.folding.childField = this.childField;
-    }
-
-    return buildFoldedResultList(this.bindings.engine, opts);
-  }
-
-  private registerDefaultResultTemplates() {
-    const content = document.createDocumentFragment();
-    const linkEl = document.createElement('atomic-result-link');
-    content.appendChild(linkEl);
-    this.resultTemplatesManager.registerTemplates({
-      content,
-      conditions: [],
+  public async initialize() {
+    this.resultListCommon = new ResultListCommon({
+      bindings: this.bindings,
+      fieldsToInclude: this.fieldsToInclude,
+      templateElements: this.host.querySelectorAll('atomic-result-template'),
```

Comment:
The `host` could be passed to the ResultListCommon and the template resolution code would be done there once